lib/lru-cache.spec.js

```diff
@@ -1,140 +1,114 @@
+    expect(current.newer).to.equal(next);
+    expect(next.older).to.equal(current);
+  }
+}
+
 describe('The LRU cache', function () {
   it('should support being called without new', function () {
     const cache = LRU(1);
-    assert(cache instanceof LRU);
+    expect(cache).to.be.an.instanceof(LRU);
   });
 
   it('should support a zero capacity', function () {
     const cache = new LRU(0);
     cache.set('key', 'value');
-    assert.equal(cache.cache.size, 0);
+    expect(cache.cache.size).to.equal(0);
   });
 
   it('should support a one capacity', function () {
     const cache = new LRU(1);
     cache.set('key1', 'value1');
-    assert.equal(cache.cache.size, 1);
-    assert.equal(cache.newest, cache.cache.get('key1'));
-    assert.equal(cache.oldest, cache.cache.get('key1'));
+    expectCacheSlots(cache, ['key1']);
     cache.set('key2', 'value2');
-    assert.equal(cache.cache.size, 1);
-    assert.equal(cache.newest, cache.cache.get('key2'));
-    assert.equal(cache.oldest, cache.cache.get('key2'));
-    assert.equal(cache.get('key1'), undefined);
-    assert.equal(cache.get('key2'), 'value2');
+    expectCacheSlots(cache, ['key2']);
+    expect(cache.get('key1')).to.be.undefined;
+    expect(cache.get('key2')).to.equal('value2');
   });
 
   it('should remove the oldest element when reaching capacity', function () {
     const cache = new LRU(2);
+
     cache.set('key1', 'value1');
     cache.set('key2', 'value2');
     cache.set('key3', 'value3');
     cache.cache.get('key1');
-    const slot2 = cache.cache.get('key2');
-    const slot3 = cache.cache.get('key3');
-    assert.equal(cache.cache.size, 2);
-    assert.equal(cache.oldest, slot2);
-    assert.equal(cache.newest, slot3);
-    assert.equal(slot2.older, null);
-    assert.equal(slot2.newer, slot3);
-    assert.equal(slot3.older, slot2);
-    assert.equal(slot3.newer, null);
-    assert.equal(cache.cache.get('key1'), undefined);
-    assert.equal(cache.get('key1'), undefined);
-    assert.equal(cache.get('key2'), 'value2');
-    assert.equal(cache.get('key3'), 'value3');
+
+    expectCacheSlots(cache, ['key2', 'key3']);
+    expect(cache.cache.get('key1')).to.be.undefined;
+    expect(cache.get('key1')).to.be.undefined;
+    expect(cache.get('key2')).to.equal('value2');
+    expect(cache.get('key3')).to.equal('value3');
   });
 
   it('should make sure that resetting a key in cache makes it newest', function () {
     const cache = new LRU(2);
+
     cache.set('key', 'value');
     cache.set('key2', 'value2');
-    assert.equal(cache.oldest, cache.cache.get('key'));
-    assert.equal(cache.newest, cache.cache.get('key2'));
+
+    expectCacheSlots(cache, ['key', 'key2']);
+
     cache.set('key', 'value');
-    assert.equal(cache.oldest, cache.cache.get('key2'));
-    assert.equal(cache.newest, cache.cache.get('key'));
+
+    expectCacheSlots(cache, ['key2', 'key']);
   });
 
   it('should make sure that getting a key in cache makes it newest', function () {
```

Comment:
I found this test quite confusing to read. I think splitting it into 3 shorter tests (one for each case in the function) would make it easier to understand.